zebra-chain/src/transaction/sighash.rs

```diff
@@ -271,14 +273,18 @@ impl<'a> SigHasher<'a> {
         writer.write_all(hash.finalize().as_ref())
     }
 
+    /// Hash the JoinSplits in a transaction.
+    ///
+    /// # Panics
+    ///
+    /// If this is called on an unsupported transaction version ([V1][Transaction::V1],
+    /// [V2][Transaction::V2] or [V5][Transaction::V5]).
```

Comment:
Should I document all methods that can panic? Or at least all methods that were updated by this PR?

The methods aren't exactly `pub`. I think most of them are `pub(super)` just so that the tests can use them. Maybe the tests should be moved into a new `sighash::tests` module?

Reply:
This code was deleted in PR #2469.